src/mats_l1_processing/instrument.py

```diff
@@ -324,11 +324,20 @@ def __init__(self, channel, calibration_file):
 
         # single event correction
         filename = calibration_data['hot_pixels']['single_events']
-        file_conn = sqlite.connect(filename)
-        self.single_event = sqlite.connect(':memory:')
-        file_conn.backup(self.single_event)
-        file_conn.close()
+        disk_conn = sqlite.connect(filename)
 
+        # Define the SQL query to select the desired data
+        query = f'''
+            SELECT * FROM SingleEvents
+            WHERE datetime BETWEEN '{start_datetime}' AND '{end_datetime}'
+            AND channel = '{self.channel}'
+        '''
+
+        # Use Pandas to read the query result into a DataFrame
+        self.single_event = pd.read_sql_query(query, disk_conn)
+        self.single_event['datetime'] = pd.to_datetime(self.single_event['datetime'])
+        # Close both database connections if they're no longer needed
+        disk_conn.close()
```

Comment:
Why treat hot pixels and single events differently? Wouldn't it be less confusing to use the same method for both?

Reply:
mainly because I'm lazy. I will add this as a new issue and implement later

Reply:
See Issue #209

Reply:
Good. I have a fairly strong preference, that this is done before I attempt to deploy again. Worst case I can probably do it myself, but I prefer to be careful about touching your code, since I'm not sure how to quality assure any changes I do make.

Reply:
ok, I will fix this then